1.regression_modelling/scripts/0.split_data.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pathlib
+
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from pycytominer import feature_select
+
+
+# ## Load in normalized profiles and concat into one dataframe
+
+# In[2]:
+
+
+# path to big drive where data is located (large)
+drive_path = pathlib.Path("../../../../media/18tbdrive/")
+
+# set path to folder in image profiling repo with the normalized profiles
+norm_path = pathlib.Path(
+    drive_path / "Github_Repositories/nuclear_speckle_image_profiling/4.preprocess_features/data/single_cell_profiles"
+).resolve(strict=True)
+
+# load all normalized parquet files
+files = norm_path.glob("*_sc_normalized.parquet")
+dfs = [pd.read_parquet(file) for file in files]
+
+# concatenate them into one data frame
+combined_df = pd.concat(dfs, ignore_index=True)
+
+# perform feature selection to drop any columns that have NaN (avoid downstream issues)
+combined_df = feature_select(
+    combined_df,
+    operation="drop_na_columns",
+    na_cutoff=0
+)
+
+# print df
+print(combined_df.shape)
+combined_df.head()
+
+
+# ## Load in filtered single-cell data from first module
+
+# In[3]:
+
+
+# load in filtered tuple dataframe
+filtering_df = pd.read_parquet(
+    pathlib.Path(
+        "../0.data_analysis_and_processing/filtered_single_cells/filtered_single_cell_profiles.parquet"
+    )
+)
+
+# print df
+print(filtering_df.shape)
+filtering_df.head()
+
+
+# ## Filter out any cells that do not matched the filtered data tuple
+
+# In[4]:
+
+
+# define the columns to match that are important for filtering
+columns_to_match = [
+    'Metadata_Plate', 'Metadata_Well', 'Metadata_Site',
+    'Metadata_Nuclei_Location_Center_X', 'Metadata_Nuclei_Location_Center_Y'
+]
+
+# ensure both dataframes have these columns
+combined_df_filtered = combined_df[columns_to_match]
+filtering_df_filtered = filtering_df[columns_to_match]
```

Comment:
This was somewhat confusing for me. Definitely semantics here and naming is hard. `filtering_df_filtered` is difficult for me to understand

Reply:
Fair point! These variables are pretty unnecessary, so I have removed them to avoid more confusion. Thanks for pointing out!